framework/test/gtest/BundleContextTest.cpp

```diff
@@ -192,4 +221,76 @@ TEST(BundleContextTest, NoSegfaultWithServiceFactory)
     }
     thread.join();
 }
+
+template <typename ServiceT>
+bool
+verifyOrdering(std::vector<cppmicroservices::ServiceReference<ServiceT>> const& refs)
+{
+    if (refs.size() > 0)
+    {
+        auto last = refs[0];
+        for (auto const& ref : refs)
+        {
+            if (!(any_cast<int>(ref.GetProperty(Constants::SERVICE_RANKING))
+                  <= any_cast<int>(last.GetProperty(Constants::SERVICE_RANKING))))
+            {
+                return false;
+            }
+
+            last = ref;
+        }
+    }
+    return true;
+}
+std::vector<ServiceProperties> props1 { { { "service.ranking", 2 } },
+                                        { { "service.ranking", 0 } },
+                                        { { "service.ranking", 4 } },
+                                        { { "service.ranking", 1 } },
+                                        { { "service.ranking", 3 } } };
+std::vector<std::pair<std::string, size_t>> vec1 {
+    {"", 5}
+};
+
+std::vector<ServiceProperties> props2 {
+    {{ "service.ranking", 2000 }, { "Key1", std::string("Val1") }},
+    {  { "service.ranking", 15 }, { "Key1", std::string("Val2") }},
+    {   { "service.ranking", 0 }, { "Key1", std::string("Val2") }},
+    {{ "service.ranking", 1506 }, { "Key2", std::string("Val1") }},
+    { { "service.ranking", 905 }, { "Key2", std::string("Val1") }}
+};
+std::vector<std::pair<std::string, size_t>> vec2 {
+    {           "", 5},
+    {"(Key1=Val*)", 3},
+    {"(Key2=Val*)", 2}
+};
+
+std::vector<std::pair<std::vector<std::pair<std::string, size_t>>, std::vector<ServiceProperties>>> parameterizedInputs {
+    {vec1, props1},
+    {vec2, props2}
+};
+
```

Comment:
all of these globals should be `const` or should be constructed in place using `::testing::Values` instead of `::testing::ValuesIn`.